web/preferences.js

```diff
@@ -108,13 +108,16 @@ class BasePreferences {
    *                    provided that the preference exists and the types match.
    */
   async set(name, value) {
-    if (typeof PDFJSDev !== "undefined" && PDFJSDev.test("MOZCENTRAL")) {
-      throw new Error("Please use `about:config` to change preferences.");
-    }
     await this.#initializedPromise;
-    AppOptions.setAll({ [name]: value }, /* prefs = */ true);
-
-    const prefs = AppOptions.getAll(OptionKind.PREFERENCE);
+    let prefs = { [name]: value };
+    AppOptions.setAll(prefs, /* prefs = */ true);
+
+    if (typeof PDFJSDev === "undefined" || !PDFJSDev.test("MOZCENTRAL")) {
+      prefs = AppOptions.getAll(OptionKind.PREFERENCE);
+    } else {
+      // In Firefox we don't need to write all the prefs but only the one that
+      // was changed.
```

Comment:
This would allow trying to write an *invalid* value, since the `AppOptions.setAll(...)`-call with the `/* prefs = */ true` argument validates the data. The following should prevent that issue:
```suggestion
      // In Firefox we don't need to write all the prefs but only the one that
      // was changed.
      prefs = { [name]: AppOptions.get(name) };
```